custom_components/thermal_comfort/config_flow.py

```diff
@@ -21,6 +27,28 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+def get_sensors_by_device_class(
+    _er: EntityRegistry,
+    _hass: HomeAssistant,
+    device_class: str,
+    include_not_in_registry: bool = False,
+) -> list:
+    """Get sensors of required class from entity registry."""
+
+    result = []
+    for d in _er.async_get_device_class_lookup(
+        {(Platform.SENSOR, device_class)}
+    ).values():
+        for e in d.values():
+            result.append(e)
```

Comment:
```python
    result = list(
        _er.async_get_device_class_lookup({(Platform.SENSOR, device_class)}).values()
    )
```
No need for a for loop here

Reply:
`async_get_device_class_lookup` returns `dict[str, dict[tuple[Any, Any], str]]` and entity_id is inside second dict. So double loop.

Reply:
ah sorry my bad, we could still go for list comprehension for performance
```python
    result = [
        entity.values()
        for entity in _er.async_get_device_class_lookup(
            {(Platform.SENSOR, device_class)}
        ).values()
    ]
```